keepassxc-browser/content/fields.js

```diff
@@ -95,9 +95,21 @@ kpxcFields.getSegmentedTOTPFields = function(inputs, combinations) {
 
     let exceptionFound = false;
 
+    // Returns true if all 6 inputs are numeric, tel or text/number with maxLength of 1
+    const allAreSegmentedFields = inputs =>
+        inputs.length === DEFAULT_SEGMENTED_TOTP_FIELDS &&
+        inputs.every(
+            i =>
+                (i.inputMode === 'numeric' && i.pattern.includes('0-9')) ||
+                ((i.type === 'text' || i.type === 'number') && i.maxLength === 1) ||
+                i.type === 'tel',
+        );
+
     const addTotpFieldsToCombination = function(inputFields, ignoreLength = false) {
-        const totpInputs = Array.from(inputFields).filter(e => e.nodeName === 'INPUT' && e.type !== 'password' && e.type !== 'hidden' && e.type !== 'submit');
-        if (totpInputs.length === DEFAULT_SEGMENTED_TOTP_FIELDS || ignoreLength) {
+        const totpInputs = Array.from(inputFields).filter(
+            e => e.nodeName === 'INPUT' && e.type !== 'password' && e.type !== 'hidden' && e.type !== 'submit',
+        );
+        if (allAreSegmentedFields(totpInputs) || ignoreLength) {
```

Comment:
if you put `ignoreLength` first then it will short circuit (if true) and avoid the inputs check. However, I don't understand the purpose of this parameter. Are we saying "don't worry about the number of segmented totp fields" or are we saying "don't worry about length==1 fields"?

Reply:
It's meant for overriding the `DEFAULT_SEGMENTED_TOTP_FIELDS`. If there's an exception, there can be e.g. 5 input fields instead of the default 6. I'll make the code more clear because this is too hard to read.